### DISCUSSION TITLE
Detect EventStream disconnetion

### DISCUSSION THREAD

**Discussion by RAPrograms** [rwf2/Rocket#2808] [Questions] [answered]
I am trying to find a way to detect when the client dissconnects an EventStream. I have been unable to find anythink on this and was wondering if anyone would be able to assist me.

**Comment by the10thWiz** [comment, marked as answer]
I believe you could use a custom Drop impl, but I'm curious what you want to do when a client disconnects?

**Reply by RAPrograms** [reply]
I want to remove some temporary user data when their connection is lost

**Reply by RAPrograms** [reply]
Thank you so much, droping works fantasticly. But how can I close the stream from the server?

**Reply by the10thWiz** [reply]
I believe you can just break out of the event stream to close the connection from the server side. I could be wrong, I haven't tested it.
